### ISSUE TITLE
Section headings in only directives are not correctly handled

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#795]
I'm not completely sure if it's a bug or just my fault,
because I'm relatively new to sphinx.

So the following does not work:

```
.. only:: html

  This is missing in html
  =======================

  This is completely missing in html

  * :ref:`genindex`
  * :ref:`modindex`
  * :ref:`search`
  * :doc:`TODO <todo>`
```

while the following does:

```
.. only:: html

  This Works
```

In http://sphinx.pocoo.org/changes.html#release-1-1-oct-9-2011 is written:
"Section headings in only directives are now correctly handled.", I don't think this is the case, I'm using Sphinx 1.2pre, compiled yesterday.

I've noticed that it works in other rst files which are included via toctree but not in index.rst. I've even created an new empty documentation via sphinx-quickstart with the same result. Is this even possible? Could somebody confirm this?

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/795
- Originally reported by: Christian Jann
- Originally created at: 2011-10-15T13:19:18.518

**Comment by shimizukawa** [comment]
_From Anonymous on 2011-11-24 07:28:17+00:00_

I'm experiencing the same issue, with Sphinx 1.1.2. To make it more concrete, I want to include a "see also" section only in the manpage version of a document:

{{{
.. only:: man

```
See Also
--------

blah blah
```

}}}

However, I notice that this problem doesn't occur if the heading level is //below// that of the previous section.

**Comment by shimizukawa** [comment]
_From Adrian Sampson on 2011-11-24 07:29:14+00:00_

(That comment was by me; I somehow didn't realize I was commenting anonymously.)

**Comment by shimizukawa** [comment]
_From [phize](https://bitbucket.org/phize) on 2011-12-17 16:16:11+00:00_

In addition, the following code also doesn't work well.

{{{
.. only:: man

```
--------
See Also
--------

blah blah
```

}}}

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2012-03-10 20:06:08+00:00_

Issue #886 was marked as a duplicate of this issue.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2012-03-10 20:21:52+00:00_

Fixes #795: explain that titles on the same level as the document heading cannot be put into only directives.

→ <<cset 729565b7a91ac2d07031aabce435b9030496934a>>
